### PR TITLE
[TTAHUB-2367] Change logic for updating fei responses on approved reports

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Currently we do NOT update FEI responses if its linked to an approved AR. 

This code change does two things:
1. **GoalFieldResponses** are ALWAYS updated regardless of onApproved report status.
2. **ActivityReportGoalFieldResponses** are ONLY updated if they are not linked to an approved report (cached as of the time of approval) all others are updated.

## How to test

- Review the code.

**Test:** 
1. Create a approved report using an FEI goal
2. Create an unapproved report using an FEI goal (using a different response than step 1)
3. Check the approved report is should still have its unique response
4. Edit the FEI goal make sure its set to the response of the unapproved goal
5. Update the response from the fei goal to something new
6. verify it was changed on the unapproved and goal
7. verify it was NOT changed on the approved report

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2367

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Review by thewatermethod — APPROVED** [review]
Looks good, verified that approved report data remains constant while everything else is updated... please wait for the CI to pass before merging :)

**Review by nvms — APPROVED** [review]
Works great!

**Pull request merged**
